File: assets/code-variants/ch12/10.java
@Test
public void displayTransactionResults_showsItemName() {
    transactionProcessor.displayTransactionResults(new User(), new Transaction("Some Item"));
    assertThat(ui.getText()).contains("You bought a Some Item");
}

@Test
public void displayTransactionResults_showsLowBalanceWarning() {
    transactionProcessor.displayTransactionResults(newUserWithBalance(LOW_BALANCE_THRESHOLD.plus(dollars(2))), new Transaction("Some Item", dollars(3)));
    assertThat(ui.getText()).contains("your balance is low");
}

File: assets/code-variants/ch12/10.py
def test_display_transaction_results_shows_item_name():
    transaction_processor.display_transaction_results(User(), Transaction("Some Item"))
    assert "You bought a Some Item" in ui.get_text()

def test_display_transaction_results_shows_low_balance_warning():
    transaction_processor.display_transaction_results(
        new_user_with_balance(LOW_BALANCE_THRESHOLD + dollars(2)), Transaction("Some Item", dollars(3)))
    assert "your balance is low" in ui.get_text()
